packages/core/src/concurrent/pool.ts

```diff
@@ -89,6 +89,14 @@ export class Pool<TResource extends Resource> implements Disposable {
     );
   }
 
+  /**
+   * Runs a task on all individual resources
+   * @param task The task to execute on each resource
+   */
+  public async runOnAllResources(task: (resource: TResource) => Promise<void>): Promise<void> {
```

Comment:
This method isn't needed anymore when you move to the new `CheckerResource` proposal.